concent_api/core/tests/test_integration_report_computed_task.py

```diff
@@ -807,75 +900,76 @@ def test_requestor_sends_reject_report_computed_task_and_then_sends_ack_report_c
+        )
 
         with freeze_time("2017-12-01 11:00:05"):
             response_3 = self.client.post(
                 reverse('core:send'),
                 data                           = serialized_reject_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
             )
 
         self.assertEqual(response_3.status_code,  202)
+        self._test_database_objects(
+            last_object_type            = message.concents.RejectReportComputedTask,
+            task_id                     = '1',
+            receive_delivered_status    = False,
+        )
 
         # STEP 4: Requestor accepts computed task via Concent
 
-        with freeze_time("2017-12-01 10:00:00"):
-            task_to_compute = message.TaskToCompute()
-
-        task_to_compute.compute_task_def = message.ComputeTaskDef()
-        task_to_compute.compute_task_def['task_id']     = '1'
-        task_to_compute.compute_task_def['deadline']    = int(dateutil.parser.parse("2017-12-01 11:00:00").timestamp())
-
-        serialized_task_to_compute      = dump(task_to_compute,             REQUESTOR_PRIVATE_KEY,  PROVIDER_PUBLIC_KEY)
-        deserialized_task_to_compute    = load(serialized_task_to_compute,  PROVIDER_PRIVATE_KEY,   REQUESTOR_PUBLIC_KEY, check_time = False)
-
-        with freeze_time("2017-12-01 11:00:05"):
-            ack_report_computed_task = message.AckReportComputedTask()
-        ack_report_computed_task.task_to_compute = deserialized_task_to_compute
-
-        serialized_ack_report_computed_task = dump(ack_report_computed_task, REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+        serialized_ack_report_computed_task = self._get_serialized_ack_report_computed_task(
+            timestamp = "2017-12-01 11:00:05",
+            ack_report_computed_task = self._get_deserialized_ack_report_computed_task(
+                subtask_id = '1',
+                task_to_compute = task_to_compute
+            ),
+            requestor_private_key = self.REQUESTOR_PRIVATE_KEY
+        )
 
         with freeze_time("2017-12-01 11:00:05"):
             response_4 = self.client.post(
                 reverse('core:send'),
                 data                           = serialized_ack_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
             )
 
-        self.assertEqual(response_4.status_code,  400)
-        self.assertIn('error', response_4.json().keys())
+        self._test_400_response(response_4)
```

Comment:
This part of the test is broken, it returns {"error": "Message timestamp too far in the future."}, but it suppose to raise:
```
raise Http400(
                "Received AckReportComputedTask but RejectReportComputedTask "
                "or another AckReportComputedTask for this task has already been submitted."
            )
```